### PR TITLE
feat: simplify python script to generate game json files

### PR DESCRIPTION
**Pull request by timotheeg**
Porting the changes I made in the other repo here: https://github.com/Mazamars312/Analogue_Pocket_Neogeo/pull/42

TLDR;
* Split romset definition from processing. romset is now in its own file `romset.json`
* Keep the list of installed games in a local variable, rather than saving it into a file `Gamelist.txt` and re-opening it all the time
* Drop the many, many duplicate functions `find_string()`
* Manipulate json structure instead of doing raw content replacements.
* Keep the template local and make copies for each game, rather than reload the template file each time.

Possible further improvement is to not have a file `template.json` at all, and keep the desired structure into the python script itself.

### CODE REVIEW

File: Assets/ng/PocketNeo.py [hunk 1]
```diff
@@ -7,8288 +7,83 @@
...
-                 
-                          # write content to second file
-                          secondfile.write(line)
-                 
-        # This for loop scans and searches each line in the file
-        # By using the input() method of fileinput module
-        for line in fileinput.input("./Mazamars312.NeoGeo/19YY.json", inplace=True):
-                 
-                 # This will replace string "GameFolder/" with "19YY" in each line
-                 line = line.replace("GameFolder/", "19YY")
```
Comment (timotheeg) [inline Assets/ng/PocketNeo.py:57]: I suspect this line was incorrect in the original script, shouldn't the code be lowercased as `19yy`? (I have left the case as is `19YY` for now in the romset definition)

File: Assets/ng/PocketNeo.py [hunk 2]
```diff
@@ -7,8288 +7,82 @@
...
-        
-                 # read content from first file
-                 for line in firstfile:
-                 
-                          # write content to second file
-                          secondfile.write(line)
-                 
-        # This for loop scans and searches each line in the file
-        # By using the input() method of fileinput module
-        for
```
Comment (timotheeg) [inline Assets/ng/PocketNeo.py:8149]: This game code also contains uppercase letters, is that normal?

File: Assets/ng/PocketNeo.py (hunk 1 above)
Comment (terminator2k2) [reply Assets/ng/PocketNeo.py:57]: yes i added that game without trying the script and noticed it didnt generate the json file yesterday, so it must use lower case.

Comment (timotheeg) [reply Assets/ng/PocketNeo.py:57]: OK, I'll update the rom code to lowercase in the PR

**Comment by timotheeg** [comment]
Just for reference, I have the dataset `Darksoft Neo Geo 2021-12-14`, and below are defined games from your list not found in it:
* `19YY`
* `CakeFighter`
* `teot`
* `lasthope`
* `nblktiger`
* `MonitorTest`
* `mslug3h`
* `mslug3c`
* `ms5plus`
* `mslug6`
* `rotdh`
* `samsh5fe`
* `samsh5pf`
* `twsoc96`
* `XenoCrisis`

File: Assets/ng/PocketNeo.py (hunk 1 above)
Comment (timotheeg) [reply Assets/ng/PocketNeo.py:57]: done in https://github.com/terminator2k2/Pocket-NeoGeo-Json-Maker/pull/1/commits/86f6b61f959c24c7304f7b11d3c897895ccf02f3

**Comment by terminator2k2** [comment]
Thank you , seems to work very well , much appreciated for your work here. The roms not in the darksoft pack can be found and converted. ms5plus is a bootleg which doesnt work with the current core but works ok in mame so i kept in there in hope that one day it works haha. This is another reason i set this up was so people didnt have json files showing for games they didnt have on their pockets. I will add your name to the credits if thats ok with you?

**Comment by timotheeg** [comment]
Thanks for merging the PR!

> I will add your name to the credits if thats ok with you?

Sure thing. Thanks for that too! 😄
